packages/kbn-eslint-rule-overrides/src/create_no_restricted_imports_override.js

```diff
@@ -0,0 +1,227 @@
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+/**
+ * @typedef {string} RestrictedImportString
+ * A string specifying a module name to restrict.
+ */
+
+/**
+ * @typedef {Object} RestrictedImportPath
+ * Configuration for restricting a specific import path.
+ * @property {string} name - The name of the module to restrict.
+ * @property {string} [message] - Custom message appended to the lint error.
+ * @property {string[]} [importNames] - Specific named imports to restrict.
+ * @property {string[]} [allowImportNames] - Named imports to allow (restricts all others).
+ */
+
+/**
+ * @typedef {Object} RestrictedImportPatternGroup
+ * Configuration for restricting imports using gitignore-style patterns.
+ * @property {string[]} group - Array of gitignore-style patterns to restrict.
+ * @property {string} [message] - Custom message.
+ * @property {string[]} [importNames] - Named imports to restrict within the group.
+ * @property {string[]} [allowImportNames] - Named imports to allow within the group.
+ * @property {boolean} [caseSensitive] - Whether the pattern is case-sensitive.
+ */
+
+/**
+ * @typedef {Object} RestrictedImportPatternRegex
+ * Configuration for restricting imports using regex patterns.
+ * @property {string} regex - Regex pattern string to restrict.
+ * @property {string} [message] - Custom message.
+ * @property {string[]} [importNames] - Named imports to restrict within the regex.
+ * @property {string[]} [allowImportNames] - Named imports to allow within the regex.
+ * @property {boolean} [caseSensitive] - Whether the regex is case-sensitive.
+ */
+
+/**
+ * @typedef {Object} RestrictedImportOptions
+ * Options for the no-restricted-imports rule.
+ * @property {Array<RestrictedImportString | RestrictedImportPath>} [paths] - List of module names or path configurations to restrict.
+ * @property {Array<string | RestrictedImportPatternGroup | RestrictedImportPatternRegex>} [patterns] - List of patterns to restrict.
+ */
+
+/**
+ * @typedef {'error' | 'warn' | 'off' | 0 | 1 | 2} RuleSeverity
+ * ESLint rule severity levels.
+ */
+
+/**
+ * @typedef {Object} CreateOverrideOptions
+ * Options for creating override configurations.
+ * @property {string} childConfigDir - Directory path for nested eslint configuration
+ * @property {Array<RestrictedImportString | RestrictedImportPath>} restrictedImports - Additional restricted imports to add.
+ */
+
+const path = require('path');
+
+const rootDir = path.resolve(__dirname, '..', '..', '..');
+
+// eslint-disable-next-line import/no-dynamic-require, @kbn/imports/no_boundary_crossing
+const rootConfig = require('../../../.eslintrc');
```

Comment:
I understand we are crossing boundaries here by hardcoding the path to root eslint config file. But I'd argue we can't avoid it unless our root .eslintrc.js lives in its own package which would understandably break a lot of IDEs (unless symlinked). But either way it's not the case so for now this looks to me like a justified tradeoff. But I am open to suggestions.